trueblocks_queries/comp_rates.sh

```diff
@@ -1,8 +1,30 @@
-#!/bin/bash
-last_month_block=$(chifra when 2022-12-13 --no_header | cut -f1)
+#!/bin/zsh
 
-chifra list 0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5 --no_header --first_block $last_month_block | cut -f2 | sort -u -n -r > last_month_blocks.txt
+# store compound cToken addresses
+eth=0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5
+btc=0xccF4429DB6322D5C611ee964527D42E5d685DD6a
 
-chifra state --call "0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5 | supplyRatePerBlock()(uint)" --file last_month_blocks.txt --no_header | awk '$5 ~ /^[1-9]/ { print $1","(($5 / 10^18 * 7200 * 365 * 100)) }' > month_results.txt
+# export all appearances
+chifra list $eth --no_header | cut -f2 | sort -u -n -r >blocksETH.txt
+chifra list $btc --no_header | cut -f2 | sort -u -n -r >blocksBTC.txt
 
-< month_results.txt jp -input csv -xy '[*][0,1]'
+# export all rates
+chifra state --call "$eth | supplyRatePerBlock()(uint)" --file blocksETH.txt --no_header | awk '$5 ~ /^[1-9]/ { print (($5 / 10^18 * 7200 * 365)) }' >resultsETH.txt
+chifra state --call "$btc | supplyRatePerBlock()(uint)" --file blocksBTC.txt --no_header | awk '$5 ~ /^[1-9]/ { print (($5 / 10^18 * 7200 * 365)) }' >resultsBTC.txt
+
+# count the number of result lines
+# this is less than the number of blocks
+# because chifra ignores "unripe" blocks
+# more than 28 blocks behind the head
```

Comment:
This comment is confusing - this makes me think you're clarifying against a race condition of some sort.